src/gui/ui/main_ui.cpp

```diff
@@ -0,0 +1,405 @@
+// Dear ImGui: standalone example application for GLFW + OpenGL 3, using programmable
+// pipeline (GLFW is a cross-platform general purpose library for handling windows,
+// inputs, OpenGL/Vulkan/Metal graphics context creation, etc.) If you are new to Dear
+// ImGui, read documentation from the docs/ folder + read the top of imgui.cpp. Read
+// online: https://github.com/ocornut/imgui/tree/master/docs
+
+// Some slight modifications
+#include "../../entity/mesh.hpp"
+#include "../../logging.hpp"
+#include "../../world.hpp"
+#include "../data_structures/frame_buffer_multisample.hpp"
+#include "../data_structures/static_mesh.hpp"
+#include "../data_structures/terrain_mesh.hpp"
+#include "../gui_logging.hpp"
+#include "../handler.hpp"
+#include "../render/quad_renderer.hpp"
+#include "../render/quad_renderer_multisample.hpp"
+#include "../render/renderer.hpp"
+#include "../render/shadow_map.hpp"
+#include "../render/sky.hpp"
+#include "../scene/controls.hpp"
+#include "../scene/scene.hpp"
+#include "../shader.hpp"
+
+#include <imgui/backends/imgui_impl_glfw.h>
+#include <imgui/backends/imgui_impl_opengl3.h>
+#include <imgui/imgui.h>
+#include <stdio.h>
+#define GL_SILENCE_DEPRECATION
+#if defined(IMGUI_IMPL_OPENGL_ES2)
+#  include <GLES2/gl2.h>
+#endif
+#include <GLFW/glfw3.h> // Will drag system OpenGL headers
+
+// [Win32] Our example includes a copy of glfw3.lib pre-compiled with VS2010 to maximize
+// ease of testing and compatibility with old VS compilers. To link with VS2010-era
+// libraries, VS2015+ requires linking with legacy_stdio_definitions.lib, which we do
+// using this pragma. Your own project should not be affected, as you are likely to link
+// with a newer binary of GLFW that is adequate for your version of Visual Studio.
+#if defined(_MSC_VER) && (_MSC_VER >= 1900) && !defined(IMGUI_DISABLE_WIN32_FUNCTIONS)
+#  pragma comment(lib, "legacy_stdio_definitions")
+#endif
+
+// This example can also compile and run with Emscripten! See 'Makefile.emscripten' for
+// details.
+#ifdef __EMSCRIPTEN__
+#  include "../libs/emscripten/emscripten_mainloop_stub.h"
+#endif
+
+namespace UI {
+
+// Main code
+int
+imguiTest(World& world) {
```

Comment:
This needs to be like 6 (static) functions

And name it something other than `test`, this func is staying.